### PR TITLE
improve lp pnl calculation

### PR DESCRIPTION
**Pull request by wakamex**
* include share reserves in LP PnL value
* add test_max_short which was written when investigating possible bug in get max short
* add `pytest.skip` to dataclasses so pytest doesn't throw a warning trying to collect tests from them
* clear matplotlib warnings
* clarify get_max_short tests

before
image

after
image

before
image

after
image
nice!

### CODE REVIEW

File: src/elfpy/wallet.py [hunk 1]
```diff
@@ -113,13 +113,13 @@ def get_state(self, market: Market) -> dict:
...
-            * market.market_state.share_price
-            * (self.lp_tokens / market.market_state.lp_reserves)
-            if self.lp_tokens > 0
-            else 0.0
-        )
+        lp_token_value = 0
+        if self.lp_tokens > 0:  # proceed further only if the agent has LP tokens
+            if market.market_state.lp_reserves > 0:  # avoid divide by zero
+                share_of_pool = self.lp_tokens / market.market_state.lp_reserves
+                pool_value = market.market_state.bond_reserves * market.spot_price  # in base
```
Comment (dpaiton) [inline src/elfpy/wallet.py:120]: can you change this to:
```
pool_value = (
    market.market_state.bond_reserves * market.spot_price  # in base
    + market.market_state.share_reserves * market.market_state.share_price  # in base
)
```

File: tests/pricing_models/calc_test_dataclasses.py [hunk 2]
```diff
@@ -4,8 +4,12 @@
 from dataclasses import dataclass
 from typing import Type, Optional
 
+from pytest import skip
+
 from elfpy.types import MarketState, Quantity, StretchedTime
 
+skip(msg="These are dataclasses used for tests, not tests themselves", allow_module_level=True)
```
Comment (dpaiton) [inline tests/pricing_models/calc_test_dataclasses.py:11]: https://media.giphy.com/media/1jkV5ifEE5EENHESRa/giphy.gif

**Review by dpaiton — APPROVED** [review]
approved -- fix notebook & comments before merging

File: tests/pricing_models/calc_test_dataclasses.py (hunk 2 above)
Comment (wakamex) [reply tests/pricing_models/calc_test_dataclasses.py:11]: 😘

**Comment by wakamex** [comment]
to suppress matplotlib warnings, i found the following:

1. remove calls to `plt.show()`. these are unnecessary in notebooks. they were used simply to suppress handles from being displayed. instead, that can be resolved by adding semicolon (`;`) to the end of the previous line. or assigning the returned handles to variables.
2. suppress the specific warning:
```
import warnings
warnings.filterwarnings( "ignore", message=".*non-GUI backend.*" )
```

**Pull request merged**